torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -38,7 +90,17 @@ py::object PyRRef::toHere() {
   } else {
     // toHere() calls python_rpc_handler which acquires GIL when UserRRef holds
     // a python object
-    IValue value = std::static_pointer_cast<UserRRef>(rref_)->toHere();
+    std::vector<IValue> rawValues =
+        std::static_pointer_cast<UserRRef>(rref_)->toHere();
+    IValue value;
+    if (rref_->isPyObj()) {
+      value = jit::toIValue(
+          PythonRpcHandler::getInstance().deserialize(
+              SerializedPyObj::fromIValues(std::move(rawValues))),
+          PyObjectType::get());
+    } else {
+      value = std::move(rawValues).front();
+    }
```

Comment:
This is a new change required to make rref_impl.cpp Python free.
Essentially, we are deferring type-specific handling to Python wrapper layer.

@mrshenli @zhaojuanmao

Reply:
This LGTM. If we need to expose C++ APIs in the future, we can come back to revisit the `toHere` API. This change should be OK as both `py_rref` and `rref_impl` are purely internal stuff.